### PR TITLE
fix(vlm): align KD distributed train step

### PR DESCRIPTION
**Pull request by khazic**
## Summary
- align the VLM KD non-PP optimizer step with the VLM finetuning distributed step
- add grad-accumulation/final-backward hooks before KD backward passes
- route gradient scaling and clipping through scale_grads_and_clip_grad_norm
- set MoE auxiliary-loss backward scale and update MoE gate bias after optimizer steps
- account for CP size in VLM KD tps_per_gpu logging

## Tests
- uv run python -m pytest -q tests/unit_tests/recipes/test_vlm_kd_distributed.py
- uv run python -m pytest -q tests/unit_tests/recipes/test_finetune_vlm_helpers.py
- uv run ruff check nemo_automodel/recipes/vlm/kd.py tests/unit_tests/recipes/test_vlm_kd_distributed.py
- uv run ruff format --check nemo_automodel/recipes/vlm/kd.py tests/unit_tests/recipes/test_vlm_kd_distributed.py

### CODE REVIEW

**Comment by khazic** [comment]
Addressed Claude's PP MoE aux-loss scaling comment in de99a1eb. KD now matches finetune behavior: PP uses num_label_tokens, non-PP uses dp_group_size(include_cp=True). The unit test now covers both branches.

**Review by HuiyingLi — DISMISSED** [review]

**Review by HuiyingLi — APPROVED** [review]
